opennms-web-api/src/main/java/org/opennms/web/enlinkd/EnLinkdElementFactory.java

```diff
@@ -932,4 +989,38 @@ private String getIpList(Set<String> ipstrings) {
         
     }
 
+    private class SnmpInterfaceCache {
+        private Map<Integer, Map<Integer, List<OnmsSnmpInterface>>> map = new HashMap<>();
+        public OnmsSnmpInterface get(int nodeId, Integer ifIdx) {
+            if (ifIdx == null) {
+                return null;
+            } else {
+                var m = map.get(nodeId);
+                if (m == null) {
+                    m = m_snmpInterfaceDao.findByNodeId(nodeId).stream().collect(Collectors.groupingBy(itf -> itf.getIfIndex()));
+                    map.put(nodeId, m);
+                }
+                var l = m.get(ifIdx);
+                return l != null && l.size() == 1 ? l.get(0) : null;
```

Comment:
Can this be simplified using `Map.computeIfAbsent`?

In addition, the check for a single element can be done while fetching. This would safe some memory. Maybe even using a different `Collector` to store a value if there are no duplicates.